osu.Game/Screens/Select/Carousel/UpdateBeatmapSetButton.cs

```diff
@@ -88,6 +92,32 @@ private void load()
                 },
             });
 
+            beatmapDownloader.DownloadFailed += _ =>
+            {
+                Hide();
+
+                // Download might have failed because BeatmapSetOnlineAvailability is DownloadDisabled
+                // We want to check that so reset online info for now, background processing will update the info
+                realm.Run(r =>
+                {
+                    var matchingSet = r.Find<BeatmapSetInfo>(beatmapSetInfo.ID);
+
+                    if (matchingSet != null)
+                    {
+                        using (var transaction = r.BeginWrite())
+                        {
+                            foreach (BeatmapInfo beatmap in matchingSet.Beatmaps)
+                            {
+                                // Background processing checks LastOnlineUpdate, that's the only field we need to null
```

Comment:
I don't understand this comment, is it trying to say that `LastOnlineUpdate` should be reset to force a background reprocess? There should be a proper method or just not have this comment in general, right now it looks as if this logic is working around the flow.